src/Symfony/Component/Messenger/Envelope.php

```diff
@@ -42,6 +48,7 @@ public function __construct(object $message, array $stamps = [])
     /**
      * Makes sure the message is in an Envelope and adds the given stamps.
      *
+     * @param TMessage         $message
      * @param StampInterface[] $stamps
      */
```

Comment:
This is wrong, the method is supposed to handle any `object`, see line 56.

I think we can make it something like this:

```suggestion
     * @template TNewMessage of object
     *
     * @param TNewMessage      $message
     * @param StampInterface[] $stamps
     *
     * @return self<TNewMessage>
     */
```